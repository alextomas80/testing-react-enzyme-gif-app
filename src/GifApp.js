import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({
	defaultCategories = ['dogs', 'the office', 'homer', 'spiderman'],
}) => {
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<AddCategory setCategories={setCategories} />
			<hr />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}
		</>
	);
};
