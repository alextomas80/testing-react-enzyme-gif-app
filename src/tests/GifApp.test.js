import React from 'react';
import { shallow } from 'enzyme';
import { GifApp } from '../GifApp';

describe('Pruebas en <GifApp />', () => {
	test('debe mostrarse correctamente', () => {
		const wrapper = shallow(<GifApp />);
		expect(wrapper).toMatchSnapshot();
	});

	test('debe de mostrar una lista de categorias', () => {
		const categories = ['Scary Movie', 'American Pie'];
		const wrapper = shallow(<GifApp defaultCategories={categories} />);

		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
