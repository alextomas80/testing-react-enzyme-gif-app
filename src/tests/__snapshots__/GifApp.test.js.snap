// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Scary Movie"
    key="Scary Movie"
  />
  <GifGrid
    category="American Pie"
    key="American Pie"
  />
</Fragment>
`;

exports[`Pruebas en <GifApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="dogs"
    key="dogs"
  />
  <GifGrid
    category="the office"
    key="the office"
  />
  <GifGrid
    category="homer"
    key="homer"
  />
  <GifGrid
    category="spiderman"
    key="spiderman"
  />
</Fragment>
`;
